node/src/components/small_network/metrics.rs

```diff
@@ -212,6 +255,17 @@ impl Metrics {
             debug!("not recording metrics, component already shut down");
         }
     }
+
+    /// Creates a set of outgoing metrics that is connected to this set of metrics.
+    pub(super) fn create_outgoing_metrics(&self) -> OutgoingMetrics {
```

Comment:
So we now have the same metrics in two places? Can it happen that the same event will update them twice b/c it will be handled on multiple layers that each can modify it?

Reply:
The metrics struct just hold a reference to a metric - it's a `Arc` around an `AtomicI64`, essentially.

The metrics are only updated inside the `outgoing` module.

Reply:
I understand that `Metrics` holds an `Arc` but it means it can also be incremented/decremented there, correct? Which means that the same metric can be changed in two places?

Reply:
Yes, but that's true for all metrics?

Reply:
Hmm, other metrics seem to be reachable only from a single place in the code.

Reply:
That's more of a coincidence :)

Reply:
Not really, at least I was doing that on purpose and thought that it was designed that way as well :) Not a blocker, just something to be aware of and watch out for.

Reply:
Fair enough!

I'll wait for Fraser to greenlight this PR still, as he was the original reviewer that had concerns.